core/src/main/java/org/elasticsearch/index/translog/BufferingTranslogWriter.java

```diff
@@ -46,6 +46,7 @@ public BufferingTranslogWriter(ShardId shardId, long generation, ChannelReferenc
 
     @Override
     public Translog.Location add(BytesReference data) throws IOException {
+        ensureOpen();
```

Comment:
this is not too much of a deal here but I can move these into the write lock block. On those methods it's really just best effort anyway. The add method has it in the right place. 

> you mean removing the ensure open? +1 on that.

I can't remove it completely I think but it's confusing and need to be used consistently. I wanna do that in a different PR since I wanna backport this one to a release branch